editor/components/rich-text/format-toolbar/positioned-at-selection.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { getOffsetParent, getRectangleFromRange } from '@wordpress/dom';
+
+/**
+ * Returns a style object for applying as `position: absolute` for an element
+ * relative to the bottom-center of the current selection. Includes `top` and
+ * `left` style properties.
+ *
+ * @return {Object} Style object.
+ */
+function getCurrentCaretPositionStyle() {
+	const selection = window.getSelection();
+
+	// Get position relative viewport.
+	const rect = getRectangleFromRange( selection.getRangeAt( 0 ) );
```

Comment:
Though improbable in the context in which we expect to use this component, should we guard against `IndexSizeError` by checking `selection.rangeCount > 0` first?

Reply:
>Though improbable in the context in which we expect to use this component, should we guard against `IndexSizeError` by checking `selection.rangeCount > 0` first?

Yep, seems very reasonable. Updated in the rebased 6eed87075747b7b79979b96b2e9c24f7b17316f0.